### PR TITLE
Bump Tested Up To Versions for 3.6 Release

### PR DESCRIPTION
**Pull request by jessy-p**
Bumps the tested to version based on any available release candidates for WordPress or WooCommerce in this or the next WCPay release cycle. Updates the minimum required version by  increasing the version by 1.

#### Changes proposed in this Pull Request
- Bumped the tested to version based on:
   - [WordPress 5.9 release date is January 25, 2022](https://make.wordpress.org/core/5-9/)
   - [WooCommerce 6.1 release date is January 11, 2022](https://developer.woocommerce.com/2021/12/28/woocommerce-6-1-rc1/)
- Updated the minimum required version
  - WordPress 5.6 to 5.7
  - WooCommerce 4.4 to 4.5
- Updated CI test versions

#### Testing instructions
- Validate the changes in `woocommerce-payments.php` and `readme.txt` 
- CI tests have been updated, should pass and cover all required compatabilities.

-------------------

- [x] Added changelog entries to both `readme.txt` and `changelog.txt` (or does not apply)
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [x] Tested on mobile (or does not apply)

**Post merge**

- [x] Added testing instructions to the [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) (or does not apply)

### CODE REVIEW

**Review by htdat — CHANGES_REQUESTED** [review]
> Validate the changes in woocommerce-payments.php and readme.txt for Tested up to

✅ This part looks good to me!

> Validate that no changes are required for **_Requires at least_**, as per the [L-2 transition timeline](https://developer.woocommerce.com/2021/05/12/woocommerce-payments-is-adopting-a-new-version-support-policy/).

I think this doc is obsolete. We need to follow this part in PaJDYF-1iC-p2 especially the underline texts:

Markup on 2022-01-04 at 11:29:17

So I think we will need to change this to 4.5 and update CI tests.

https://github.com/Automattic/woocommerce-payments/blob/f582049892d773aeb781b311447f5cf207af514c/woocommerce-payments.php#L11

**Comment by htdat** [comment]
> So I think we will need to change this to 4.5 and update CI tests.

I've spotted a few CI tests to update: 

- Change 4.4.0 to 4.5.0:
https://github.com/Automattic/woocommerce-payments/blob/f582049892d773aeb781b311447f5cf207af514c/.github/workflows/e2e-test.yml#L38-L39

- Remove 4.4.0 for matrix. For the edge case, change 4.4.0 to 4.5.0 - may we need to update WP version from 5.4 to 5.5 too? 
https://github.com/Automattic/woocommerce-payments/blob/f582049892d773aeb781b311447f5cf207af514c/.github/workflows/compatibility.yml#L13-L21

--- 

Also, this would change to 5.6.0 as the current L-2 of Woo core, which would have been done in the previous WCPay release.

https://github.com/Automattic/woocommerce-payments/blob/f582049892d773aeb781b311447f5cf207af514c/.github/workflows/php-lint-test.yml#L8-L9

**Review by jrodger — DISMISSED** [review]
Thanks for tackling this new process @jessy-p. The changes look good to me and I agree with the points that @htdat raised. I think we're good to merge this once those are addressed.

**Comment by jessy-p** [comment]
Addressed all review comments except  update wordpress from 5.4 to 5.6  in oldest dependency compatibility because of [ an issue](p1641279509415000/1641232729.391900-slack-CGGCLBN58)

File: phpcs-compat.xml.dist [hunk 1]
```diff
@@ -15,7 +15,7 @@
 	<exclude-pattern>*\.(?!php$)</exclude-pattern>
 
 	<!-- Configs -->
-	<config name="minimum_supported_wp_version" value="5.6" />
+	<config name="minimum_supported_wp_version" value="5.7" />
```
Comment (htdat) [inline phpcs-compat.xml.dist:18]: WCPay 3.6 is released on Jan 19 while WP core 5.9 is released later on Jan 25. I think it would be better to update this minimum supported version for WP (here and other places) in WCPay 3.7?

File: .github/workflows/compatibility.yml [hunk 2]
```diff
@@ -11,12 +11,12 @@ jobs:
...
       matrix:
-        woocommerce: [ '4.4.0', '4.5.0', '4.6.0', '4.7.0', '4.8.0', '4.9.0', '5.0.0', '5.1.0', '5.2.0', '5.3.0', '5.4.0', '5.5.0', '5.6.0', '5.7.0', '5.8.0', '5.9.0', '6.0.0' ]
+        woocommerce: [ '4.5.0', '4.6.0', '4.7.0', '4.8.0', '4.9.0', '5.0.0', '5.1.0', '5.2.0', '5.3.0', '5.4.0', '5.5.0', '5.6.0', '5.7.0', '5.8.0', '5.9.0', '6.0.0', '6.1.0-rc.1']
         wordpress:   [ 'latest' ]
         php:         [ '7.4' ]
         include:
           # Edge case: oldest dependencies compatibility
-          - woocommerce: '4.4.0'
+          - woocommerce: '4.5.0'
             wordpress:   '5.4'
```
Comment (htdat) [inline .github/workflows/compatibility.yml:20]: I think it's still good to update this WP version from 5.4 to 5.6. That's because, in practice, WCPay is requiring WP version >= 5.6.

Comment (jessy-p) [reply .github/workflows/compatibility.yml:20]: But the E2E tests are failing. I think we need to fix that first then.

Comment (htdat) [reply .github/workflows/compatibility.yml:20]: > I think we need to fix that first then ( on a separate PR ).

👍 Agreed!

Comment (htdat) [reply .github/workflows/compatibility.yml:20]: Let me spin up another PR as I had some ideas about it in this test PR https://github.com/Automattic/woocommerce-payments/pull/3346

Comment (htdat) [reply .github/workflows/compatibility.yml:20]: > Let me spin up another PR as I had some ideas about it in this test PR #3346

Here is the PR https://github.com/Automattic/woocommerce-payments/pull/3597

File: phpcs-compat.xml.dist (hunk 1 above)
Comment (jessy-p) [reply phpcs-compat.xml.dist]: Fixed

**Comment by jessy-p** [comment]
>Addressed all review comments except update wordpress from 5.4 to 5.6 in oldest dependency compatibility 
Fixed this as well.

File: .github/workflows/php-lint-test.yml [hunk 3]
```diff
@@ -4,8 +4,8 @@ on:
   pull_request
 
 env:
+  WC_VERSION: 5.6.0  # the min supported version as per L-2 policy
```
Comment (htdat) [inline .github/workflows/php-lint-test.yml:7]: Nitpick: may only change the version number rather than change its line?

File: readme.txt [hunk 4]
```diff
@@ -1,8 +1,8 @@
 === WooCommerce Payments ===
 Contributors: woocommerce, automattic
 Tags: woocommerce, payment, payment request, credit card, automattic
-Requires at least: 5.6
-Tested up to: 5.8.2
+Requires at least: 5.7
```
Comment (htdat) [inline readme.txt:4]: May revert this change back to 5.6?

File: changelog.txt [hunk 5]
```diff
@@ -1,6 +1,8 @@
 *** WooCommerce Payments Changelog ***
 
 = 3.6.0 - 2022-xx-xx =
+* Update - Bump minimum required version of WooCommerce from 4.4 to 4.5.
+* Update - Bump minimum required version of WordPress from 5.6 to 5.7.
```
Comment (htdat) [inline changelog.txt:5]: May remove this line?

File: readme.txt [hunk 6]
```diff
@@ -38,7 +38,7 @@ Our global support team is available to answer questions you may have about WooC
 
 = Requirements =
 
-* WordPress 5.6 or newer.
+* WordPress 5.7 or newer.
```
Comment (htdat) [inline readme.txt:41]: Revert this change too?

File: readme.txt [hunk 7]
```diff
@@ -99,6 +99,8 @@ Please note that our support for the checkout block is still experimental and th
 == Changelog ==
 
 = 3.6.0 - 2022-xx-xx =
+* Update - Bump minimum required version of WooCommerce from 4.4 to 4.5.
+* Update - Bump minimum required version of WordPress from 5.6 to 5.7.
```
Comment (htdat) [inline readme.txt:103]: May remove this line too?

File: woocommerce-payments.php [hunk 8]
```diff
@@ -8,9 +8,9 @@
  * Woo: 5278104:bf3cf30871604e15eec560c962593c1f
  * Text Domain: woocommerce-payments
  * Domain Path: /languages
- * WC requires at least: 4.4
- * WC tested up to: 6.0.0
- * Requires at least: 5.6
+ * WC requires at least: 4.5
+ * WC tested up to: 6.1
+ * Requires at least: 5.7
```
Comment (htdat) [inline woocommerce-payments.php:13]: May revert this change back to 5.6 too?

**Review by htdat — COMMENTED** [review]
LGTM! After reverting back changes related to WP minimum version, then it's ready to 🚢

File: .github/workflows/php-lint-test.yml (hunk 3 above)
Comment (jessy-p) [reply .github/workflows/php-lint-test.yml]: Fixed

File: changelog.txt (hunk 5 above)
Comment (jessy-p) [reply changelog.txt]: Removed

**Review by htdat — APPROVED** [review]
